File: GUILib/src/space/kiibou/event/MouseEventModifier.java
package space.kiibou.event;

import processing.event.MouseEvent;

import java.util.EnumSet;

public enum MouseEventModifier {
    SHIFT, CTRL, META, ALT;

    public static EnumSet<MouseEventModifier> fromProcessingEvent(MouseEvent event) {
        EnumSet<MouseEventModifier> modifiers = EnumSet.noneOf(MouseEventModifier.class);
        if (event.isAltDown()) modifiers.add(ALT);
        if (event.isControlDown()) modifiers.add(CTRL);
        if (event.isMetaDown()) modifiers.add(META);
        if (event.isShiftDown()) modifiers.add(SHIFT);
        return modifiers;
    }
}

File: GUILib/src/space/kiibou/event/MouseEventModifier.kt
package space.kiibou.event

import processing.event.MouseEvent
import java.util.*

enum class MouseEventModifier {
    SHIFT, CTRL, META, ALT;

    companion object {
        fun fromProcessingEvent(event: MouseEvent): EnumSet<MouseEventModifier> {
            val modifiers = EnumSet.noneOf(MouseEventModifier::class.java)
            if (event.isAltDown) modifiers.add(ALT)
            if (event.isControlDown) modifiers.add(CTRL)
            if (event.isMetaDown) modifiers.add(META)
            if (event.isShiftDown) modifiers.add(SHIFT)
            return modifiers
        }
    }
}